code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +40,184 @@ internal sealed class AEPPushTemplate {
-        smallIconColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.SMALL_ICON_COLOR)
-        notificationBackgroundColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.NOTIFICATION_BACKGROUND_COLOR)
-        isFromIntent = true
+    internal open fun createIntent(action: String): Intent {
+        val intent = Intent(action)
+        intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
+
+        // add the notification payload version
+        intent.putExtra(PushPayloadKeys.VERSION, payloadVersion?.toString())
+
+        // add the notification text information
+        intent.putExtra(PushPayloadKeys.TITLE, title)
+        intent.putExtra(PushPayloadKeys.BODY, body)
+        intent.putExtra(PushPayloadKeys.EXPANDED_BODY_TEXT, expandedBodyText)
+        intent.putExtra(PushPayloadKeys.TICKER, ticker)
+
+        // add the template type
+        intent.putExtra(PushPayloadKeys.TEMPLATE_TYPE, templateType?.value)
+
+        // add the basic media information
+        intent.putExtra(PushPayloadKeys.IMAGE_URL, imageUrl)
+
+        // add the notification action information
+        intent.putExtra(PushPayloadKeys.ACTION_URI, actionUri)
+        intent.putExtra(PushPayloadKeys.ACTION_TYPE, actionType?.name)
+
+        // add the notification icon information
+        intent.putExtra(PushPayloadKeys.SMALL_ICON, smallIcon)
+        intent.putExtra(PushPayloadKeys.LARGE_ICON, largeIcon)
+
+        // add the notification color data
+        intent.putExtra(PushPayloadKeys.TITLE_TEXT_COLOR, titleTextColor)
+        intent.putExtra(PushPayloadKeys.BODY_TEXT_COLOR, bodyTextColor)
+        intent.putExtra(PushPayloadKeys.BACKGROUND_COLOR, backgroundColor)
+        intent.putExtra(PushPayloadKeys.SMALL_ICON_COLOR, smallIconColor)
+
+        // add other notification properties
+        intent.putExtra(PushPayloadKeys.TAG, tag)
+        intent.putExtra(PushPayloadKeys.SOUND, sound)
+        intent.putExtra(PushPayloadKeys.BADGE_COUNT, badgeCount.toString())
+        intent.putExtra(PushPayloadKeys.STICKY, isNotificationSticky.toString())
+
+        // add notification priority and visibility
+        intent.putExtra(PushPayloadKeys.PRIORITY, priorityString)
+        intent.putExtra(PushPayloadKeys.VISIBILITY, visibilityString)
+
+        return intent
     }
 
-    fun getNotificationImportance(): Int {
-        notificationImportance?.let {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) return notificationImportance as Int
-        }
-        return NotificationManager.IMPORTANCE_DEFAULT
+    fun getNotificationVisibility(): Int {
+        return NotificationVisibility.getNotificationCompatVisibilityFromString(visibilityString)
     }
 
-    fun getNotificationVisibility(): Int {
-        return notificationVisibility ?: NotificationCompat.VISIBILITY_PRIVATE
+    fun getNotificationPriority(): Int {
```

Comment:
Same comment here as `getNotificationVisibility`

Reply:
Like mentioned above this method would still be valid..  It will be the only way to get NotificationPriority in Int
Will come back to cleaning up this value in my next PR.